packages/remix-dev/config.ts

```diff
@@ -353,12 +353,19 @@ export async function readConfig(
       path.join("public", "build")
   );
 
-  if (typeof appConfig.devServerPort !== "number") {
-    appConfig.devServerPort = Number(
-      process.env.REMIX_DEV_SERVER_WS_PORT || 8002
-    );
+  let devServerPort: number;
+
+  // check if the port is already in use as this file is reloaded after a dep install or config change
+  if (process.env.REMIX_DEV_SERVER_WS_PORT) {
+    try {
+      devServerPort = Number(process.env.REMIX_DEV_SERVER_WS_PORT);
+    } catch (error: unknown) {
```

Comment:
I'm unsure if `Number` ever throws an error. In most cases it will return `NaN` instead. 

In this part of the code, it would mean that someone explicitly set the env var to an invalid value. If it wasn't set explicitly, then it would either 1) fall through to `getPort` on the initial load, or 2) have been set to the output of `getPort` on a previous load. 

Is `getPort` as default the right thing to do if someone explicitly sets the env var to an invalid value? (Genuinely curious as to the philosophy of the Remix team.)

Reply:
yup, you're right on the NaN, Mondays amirite 😅. i'll wait for @kentcdodds' thoughts on the default behavior, i could see throwing an error as the desired behavior

Reply:
Awesome. :) Thanks so much for working on this. I'm excited to see LiveReload working again!